workers/unity/Assets/Playground/Scripts/UI/ResizedGui.cs

```diff
@@ -0,0 +1,69 @@
+﻿using System;
+using UnityEngine;
+
+namespace Playground
+{
+    internal struct ResizedGui : IDisposable
+    {
+        private readonly GUIStyle[] styles;
+        private readonly Font[] fontCache;
+        private readonly int[] fontSizeCache;
+
+        public ResizedGui(Font font, float screenWidthFontRatio, params GUIStyle[] skins)
+        {
+            styles = skins;
+
+            fontCache = new Font[skins.Length];
+            fontSizeCache = new int[skins.Length];
+
+            var actualFontSize = (int) (Screen.width * screenWidthFontRatio);
+
+            for (var i = 0; i < styles.Length; i++)
+            {
+                var guiStyle = styles[i];
+                fontCache[i] = guiStyle.font;
+                fontSizeCache[i] = guiStyle.fontSize;
+
+                guiStyle.font = font;
+                guiStyle.fontSize = actualFontSize;
+            }
+        }
+
+        public void Dispose()
+        {
+            for (var i = 0; i < styles.Length; i++)
+            {
+                var guiStyle = styles[i];
+                guiStyle.font = fontCache[i];
+                guiStyle.fontSize = fontSizeCache[i];
+            }
+        }
+
+        private static readonly GUIContent TempGuiContent = new GUIContent();
+
+        public static void Label(string labelText)
+        {
+            TempGuiContent.text = labelText;
+            GUILayout.Label(labelText, GUILayout.Height(GUI.skin.label.CalcSize(TempGuiContent).y));
```

Comment:
should we return here as well_ just to stay consistent

Reply:
Can't, labels are void too